src/lib/util/validateOrigin.test.ts

```diff
@@ -0,0 +1,21 @@
+import { validateOrigin } from './validateOrigin';
+
+test('validateOrigin', () => {
+    expect(validateOrigin('')).toEqual(false);
+    expect(validateOrigin(' ')).toEqual(false);
+    expect(validateOrigin('a')).toEqual(false);
+    expect(validateOrigin('**')).toEqual(false);
+    expect(validateOrigin('localhost')).toEqual(false);
+    expect(validateOrigin('localhost:8080')).toEqual(false);
+    expect(validateOrigin('//localhost:8080')).toEqual(false);
+    expect(validateOrigin('http://localhost/')).toEqual(false);
+    expect(validateOrigin('http://localhost/a')).toEqual(false);
+    expect(validateOrigin('https://example.com/a')).toEqual(false);
+    expect(validateOrigin('https://example.com ')).toEqual(false);
+
+    expect(validateOrigin('*')).toEqual(true);
```

Comment:
is `*.localhost` valid? or `https://*.example.com`?

Reply:
Nope, for CORS we have to use `*` or a full origin. Would be nice though!

Reply:
Could be added to as expected to fail

Reply:
Yeah, good idea!